### PR TITLE
MOSIP-31518, MOSIP-31850 : Added resident role for partner api

### PR DESCRIPTION
**Pull request by GOKULRAJ136**

### CODE REVIEW

File: partner/pms-common/src/main/java/io/mosip/pms/common/helper/SearchHelper.java [hunk 1]
```diff
@@ -538,7 +538,8 @@ private SearchDto addPartnerFilter(SearchDto searchDto, String partnerIdColumn)
 	}
 	
 	public boolean isLoggedInUserFilterRequired() {
-		return !(UserDetailUtil.getLoggedInUserDetails().getAuthorities().stream()
-				.anyMatch(r -> r.getAuthority().equalsIgnoreCase("ROLE_" + partnerAdminRole)));
+		List<String> roles = Arrays.asList(partnerAdminRole.split(","));
```
Comment (dhanendra06) [inline partner/pms-common/src/main/java/io/mosip/pms/common/helper/SearchHelper.java:541]: don't split with commas please use @value and store in a variable with List<String>dataType not required to split with commas

**Review by dhanendra06 — CHANGES_REQUESTED** [review]

File: partner/pms-common/src/main/java/io/mosip/pms/common/helper/SearchHelper.java (hunk 1 above)
Comment (GOKULRAJ136) [reply partner/pms-common/src/main/java/io/mosip/pms/common/helper/SearchHelper.java]: changed as per the requirement

**Review by aranaravi — APPROVED** [review]

File: partner/pms-common/src/main/java/io/mosip/pms/common/helper/SearchHelper.java
Comment (ase-101) [inline partner/pms-common/src/main/java/io/mosip/pms/common/helper/SearchHelper.java:1]: @dhanendra06 let us check why does resident need access to this endpoint, if at all this information is required in resident portal should we not use resident client ID and secret?

cc: @LoganathanSekar7627

Comment (dhanendra06) [reply partner/pms-common/src/main/java/io/mosip/pms/common/helper/SearchHelper.java:1]: I have checked the issue @ase-101 resident needs this access to get the organization name based on partner to send email and resident sending request based on resident client and secret but we have restricted the role access so to add RESIDENT role in property we are making this change.

**Review by LoganathanSekar7627 — APPROVED** [review]

**Review by ckm007 — APPROVED** [review]

**Pull request merged**